tests/unit/backend/layers/api/test_curation_api.py

```diff
@@ -0,0 +1,1229 @@
+import copy
+import json
+from dataclasses import asdict
+from unittest.mock import Mock, patch
+
+from backend.common.corpora_orm import CollectionVisibility
+from backend.common.providers.crossref_provider import CrossrefDOINotFoundException
+from backend.common.utils.api_key import generate
+from backend.layers.common.entities import (
+    CollectionVersion,
+    DatasetProcessingStatus,
+    DatasetStatusKey,
+    DatasetUploadStatus,
+    DatasetValidationStatus,
+    OntologyTermId,
+)
+from backend.portal.api.curation.v1.curation.collections.common import EntityColumns
+from unit.backend.api_server.base_api_test import BaseAuthAPITest
+from unit.backend.fixtures.mock_aws_test_case import CorporaTestCaseUsingMockAWS
+from unit.backend.layers.common.base_api_test import DatasetArtifactUpdate, DatasetStatusUpdate, NewBaseTest
+
+
+class TestAsset(BaseAuthAPITest, CorporaTestCaseUsingMockAWS):
+    def setUp(self):
+        # Needed for proper setUp resolution in multiple inheritance
+        super().setUp()
+        self.test_dataset_id = "test_dataset_id"
+
+    def test__get_dataset_asset__OK(self):
+        bucket = self.CORPORA_TEST_CONFIG["bucket_name"]
+        s3_file_name = "test_s3_uri.h5ad"
+        content = "Hello world!"
+        self.create_s3_object(s3_file_name, bucket, content=content)
+
+        expected_body = [dict(filename="test_filename", filesize=len(content), filetype="H5AD")]
+
+        response = self.app.get(
+            f"/curation/v1/collections/test_collection_id/datasets/{self.test_dataset_id}/assets",
+        )
+        self.assertEqual(200, response.status_code)
+        actual_body = response.json
+        presign_url = actual_body[0].pop("presigned_url")
+        self.assertIsNotNone(presign_url)
+        self.assertEqual(expected_body, actual_body)
+
+    def test__get_dataset_asset__file_error(self):
+        expected_body = [dict(filename="test_filename", filesize=-1, filetype="H5AD")]
+
+        response = self.app.get(
+            f"/curation/v1/collections/test_collection_id/datasets/{self.test_dataset_id}/assets",
+        )
+        self.assertEqual(202, response.status_code)
+        actual_body = response.json
+        presign_url = actual_body[0].pop("presigned_url", None)
+        self.assertIsNone(presign_url)
+        self.assertEqual(expected_body, actual_body)
+
+    def test__get_dataset_asset__dataset_NOT_FOUND(self):
+        bad_id = "bad_id"
+
+        response = self.app.get(
+            f"/curation/v1/collections/test_collection_id/datasets/{bad_id}/assets",
+        )
+        self.assertEqual(404, response.status_code)
+        actual_body = response.json
+        self.assertEqual("Dataset not found.", actual_body["detail"])
+
+    def test__get_dataset_asset__collection_dataset_NOT_FOUND(self):
+        """Return Not found when the dataset is not part of the collection"""
+        bad_id = "bad_id"
```

Comment:
todo: generate UUID to fit integration tests (db constraint is UUID for ID fields)